packages/importer/src/random-data.ts

```diff
@@ -1,71 +1,75 @@
 import { fakerEN as faker } from '@faker-js/faker';
 import { Schemas } from '@xata.io/client';
+import { z } from 'zod';
 
 export function generateRandomData(table: Schemas.Table, size: number) {
   const records: Record<string, unknown>[] = [];
 
   for (let index = 0; index < size; index++) {
-    records.push(randomRecord(table.columns));
+    // TODO fix type
+    records.push(randomRecord(table.columns as any));
   }
 
   return records;
 }
 
-function randomRecord(columns: Schemas.Column[]) {
+function randomRecord(columns: (Schemas.Column & { comment?: string; pgType: string })[]) {
   const record: Record<string, unknown> = {};
   for (const column of columns) {
+    // TODO column should contain pgType field from schema
     record[column.name] = randomData(column);
   }
   return record;
 }
 
-function randomData(column: Schemas.Column) {
-  switch (column.type) {
-    case 'text':
-      return faker.lorem.paragraphs(rand(2, 3));
-    case 'email':
-      return faker.internet.email({ provider: 'acme.pets' });
+function randomData(column: Schemas.Column & { comment?: string; pgType: string }) {
+  const columnCommentType = narrowStringType(column.comment);
+  // Note that this is a best effort and seeding may fail for invalid Xata columns
+  // that are foreign keys, or have constraints such as length attached to them.
+  switch (column.pgType) {
+    case 'boolean':
+    case 'bool':
+      return rand(0, 1) === 1;
+    case 'bigint':
+    case 'int8':
+    case 'integer':
     case 'int':
+    case 'int4':
+    case 'smallint':
       return rand(1, 100);
-    case 'float':
+    case 'double precision':
+    case 'float8':
+    case 'real':
       return rand(1, 10000) / rand(1, 100);
-    case 'bool':
-      return rand(0, 1) === 1;
-    case 'multiple':
-      return faker.word.words(rand(1, 3)).split(' ');
-    case 'string':
-      return randomString(column.name);
-    case 'datetime':
+    case 'text':
+    case 'varchar':
+    case 'character varying':
+      if (columnCommentType === 'email') return faker.internet.email({ provider: 'acme.pets' });
+      if (column.type === 'link') return undefined;
```

Comment:
I wonder if I should filter this out in the client as well, like I do with invalid links